handler/src/main/java/io/netty/handler/timeout/IdleStateEvent.java

```diff
@@ -31,6 +32,7 @@
 
     private final IdleState state;
     private final boolean first;
+    private String strVal;
```

Comment:
You probably don't need to cache this, since it's unlikely to be reused.

Reply:
I agree with @carl-mastrangelo ... @ursaj can you remove this  ?

Reply:
you both are mistaken in the way, how this class is used. there are only 6 instances of this class in runtime:

    public class IdleStateEvent {
        public static final IdleStateEvent FIRST_READER_IDLE_STATE_EVENT = new IdleStateEvent(IdleState.READER_IDLE, true);
        public static final IdleStateEvent READER_IDLE_STATE_EVENT = new IdleStateEvent(IdleState.READER_IDLE, false);
        public static final IdleStateEvent FIRST_WRITER_IDLE_STATE_EVENT = new IdleStateEvent(IdleState.WRITER_IDLE, true);
        public static final IdleStateEvent WRITER_IDLE_STATE_EVENT = new IdleStateEvent(IdleState.WRITER_IDLE, false);
        public static final IdleStateEvent FIRST_ALL_IDLE_STATE_EVENT = new IdleStateEvent(IdleState.ALL_IDLE, true);
        public static final IdleStateEvent ALL_IDLE_STATE_EVENT = new IdleStateEvent(IdleState.ALL_IDLE, false);

So cached string value is used countless times.

Reply:
The same applies to inherited instances (if any):

* either they are used in the same shared model, so this optimisation is useful;
* or they are created, used and GC-ed in a short while, so there is no harm in this optimisation.

Reply:
I disagree here.   If it is the case that the same idle events are reused heavily, then it stands to reason that they are shared by multiple event loops.   That makes this code racy and will trigger the race detector.

Reply:
@carl-mastrangelo generate strings on each invocation doesn't look good either. any ideas or examples how it should look like?

Reply:
ah, I see. you've already fixed it in PR #9705